juriscraper/opinions/united_states/state/illappct.py

```diff
@@ -28,9 +33,15 @@ def _get_docket_numbers(self) -> List[str]:
         """
         dockets_numbers = []
         for case in self.cases:
-            m = re.search(self.docket_re, case["neutral_citation"])
-            raw_docket = m.group("docket")
-            district = m.group("district")
-            docket = f"{district}-{raw_docket[0:2]}-{raw_docket[2:]}"
-            dockets_numbers.append(docket)
+            match = re.search(self.docket_re, case["neutral_citation"])
+            if match:
+                raw_docket = match.group("docket")
+                district = match.group("district")
+                docket = f"{district}-{raw_docket[0:2]}-{raw_docket[2:]}"
+                dockets_numbers.append(docket)
+            else:
+                logger.info(f"Could not find docket for case: '{case}'")
+                raise InsanityException(
+                    f"Could not find docket for case: '{case}'"
+                )
```

Comment:
This just crashes the entire scraper right? 

An insanity exception probably isn't the right error ( I think).  That was more important before we started using opinionsitelinear - it was about the other pattern where you scraped 10 names but only 8 docket numbers.  That would be the use case for the insanity exception.  Here we would want to throw something like 

logger.warning(f"No docket number extracted ..."). and continue to the next row.  This lets us know that something didn't work for a row, but continue scraping the rest of the page.

Reply:
I'm also unsure of `InsanityException` is the best approach here. However, these are my reasons behind this decision:

- Method `_get_docket_numbers(self) -> List[str]` returns a list with all the docket attributes for all scraped case objects, not just for an individual case.
- An empty docket attribute might cause problems when **CourtListener** registers the opinion information.
- Although the original [Sentry report](https://sentry.io/organizations/freelawproject/issues/2892226059/?referrer=github_integration) only has 2 registered events (and I suspect it was due to a connection error that caused that the page didn't loaded complete), I think is best to check that each case citation includes a value, but also to raise an error in case the value of the citation no longer matches the expected pattern.

If it is an exageration to raise this exception in this method, just tell me and I'll delete it.

Reply:
I was suggesting that you log the warning and use `continue`

Reply:
Done